test/parser/parser_test.cpp

```diff
@@ -99,6 +99,42 @@ TEST_F(ParserTestBase, CreateFunctionTest) {
       "BEGIN RETURN i + 1; END; $$ "
       "LANGUAGE plpgsql;";
   stmts = pgparser.BuildParseTree(query);
+
+  query =
+      "CREATE OR REPLACE FUNCTION return_varchar ("
+      " i VARCHAR"
+      " )"
+      " RETURNS VARCHAR AS $$ "
+      "BEGIN RETURN 'foo'; END; $$ "
+      "LANGUAGE plpgsql;";
+  stmts = pgparser.BuildParseTree(query);
+
+  query =
+      "CREATE OR REPLACE FUNCTION return_text ("
+      " i TEXT"
+      " )"
+      " RETURNS TEXT AS $$ "
+      "BEGIN RETURN 'foo'; END; $$ "
+      "LANGUAGE plpgsql;";
+  stmts = pgparser.BuildParseTree(query);
+
+  query =
+      "CREATE OR REPLACE FUNCTION return_bool ("
+      " i BOOLEAN"
```

Comment:
Side question:
While taking a look again at these tests, I found that [create_function_statement.h](https://github.com/cmu-db/terrier/blob/master/src/include/parser/create_function_statement.h#L38) as two types for boolean. `Parameter::Datatype::BOOL` and `Parameter::Datatype::BOOLEAN`.
At the moment will go with `Parameter::Datatype::BOOLEAN` in the test and will update accordingly in the future.